src/Uno.UI/UI/Xaml/Shapes/Shape.Skia.cs

```diff
@@ -0,0 +1,174 @@
+		}
+
+		private void InitCommonShapeProperties() { }
+
+		protected override Size MeasureOverride(Size availableSize)
+		{
+			var geometrySource = GetGeometry(availableSize);
+
+			return new Size(geometrySource.Geometry.Bounds.Width, geometrySource.Geometry.Bounds.Height);
+		}
+
+		protected override Size ArrangeOverride(Size finalSize)
+		{
+			var area = new Rect(0, 0, finalSize.Width, finalSize.Height);
+
+			switch (Stretch)
+			{
+				default:
+				case Stretch.None:
+					break;
+				case Stretch.Fill:
+					area = new Rect(0, 0, finalSize.Width, finalSize.Height);
+					break;
+				case Stretch.Uniform:
+					area = (area.Height > area.Width)
+						? (new Rect((float)area.X, (float)area.Y, (float)area.Width, (float)area.Width))
+						: (new Rect((float)area.X, (float)area.Y, (float)area.Height, (float)area.Height));
+					break;
+				case Stretch.UniformToFill:
+					area = (area.Height > area.Width)
+						? (new Rect((float)area.X, (float)area.Y, (float)area.Height, (float)area.Height))
+						: (new Rect((float)area.X, (float)area.Y, (float)area.Width, (float)area.Width));
+					break;
+			}
+
+			var shrinkValue = -ActualStrokeThickness / 2;
+			if (area != Rect.Empty)
+			{
+				area.Inflate(shrinkValue, shrinkValue);
+			}
+
+			var geometrySource = GetGeometry(finalSize);
+			var compositionPath = new CompositionPath(geometrySource);
+			var pathGeometry = Visual.Compositor.CreatePathGeometry();
+			pathGeometry.Path = compositionPath;
+
+			_pathSpriteShape = Visual.Compositor.CreateSpriteShape(pathGeometry);
+
+			_rectangleVisual.Shapes.Clear();
+			_rectangleVisual.Shapes.Add(_pathSpriteShape);
+
+			UpdateFill();
+			UpdateStroke();
+			UpdateStrokeThickness();
+			return finalSize; // geometrySource.Geometry.Bounds.Size.ToSize();
+		}
+
+		partial void OnStrokeUpdatedPartial()
+		{
+			UpdateStroke();
+
+			InvalidateMeasure();
+		}
+
+
+		partial void OnFillUpdatedPartial()
+		{
+			UpdateFill();
+
+			InvalidateMeasure();
+		}
+
+		partial void OnStrokeThicknessUpdatedPartial()
+		{
+			if (StrokeThickness > 0 && StrokeThickness < 1)
+			{
+				StrokeThickness = 1;
+				return;
+			}
+
+			UpdateStrokeThickness();
+
+			InvalidateMeasure();
+		}
+
+		partial void OnStrokeDashArrayUpdatedPartial()
+		{
+			// _rectangleLayer.LineDashPattern = newValue.Safe().Select(d => new NSNumber(d)).ToArray();
+
+			InvalidateMeasure();
+		}
+
+		partial void OnStretchUpdatedPartial()
+		{
+			InvalidateMeasure();
+		}
+
+		private void UpdateFill()
+		{
+			if (_pathSpriteShape != null)
+			{
+				_fillSubscription.Disposable = null;
+
+				_pathSpriteShape.FillBrush = null;
+
+				var scbFill = Fill as SolidColorBrush;
+				var lgbFill = Fill as LinearGradientBrush;
+				if (scbFill != null)
+				{
+					_fillSubscription.Disposable =
+						Brush.AssignAndObserveBrush(scbFill, c => _pathSpriteShape.FillBrush = Visual.Compositor.CreateColorBrush(scbFill.Color));
+				}
+				else if (lgbFill != null)
+				{
```

Comment:
Codacy found an issue: [Either remove or fill this block of code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5872302)